src/xmlparser/OrderedObjParser.js

```diff
@@ -280,6 +280,7 @@ const parseXml = function(xmlData) {
       }else {//Opening tag
         let result = readTagExp(xmlData,i, this.options.removeNSPrefix);
         let tagName= result.tagName;
+        const tagNameWithNamespace = result.tagNamespace ? result.tagNamespace + ":" + result.tagName : result.tagName;
```

Comment:
> The original tagName variable can be renamed accordingly.

Pehaps the new name for that variable should be `tagNameWithoutNamespace`

> Moreover, it would be good if this code is present in readTagExp so reduce an extra operation of comparison and concatenation.

I'll see what I can do

Reply:
I've updated this PR, and I think the approach make sense and hopefully addresses your concerns.

`tagName` is the tag name. If `removeNSPrefix`, then `tagName` never has a namespace; otherwise, it will have a namespace if there is a namespace.

`tagNameWithNamespace` is the tag name with namespace if there is a namespace. It's value doesn't depend on `removeNSPrefix`.

The work is done in `readTagExp` and it avoids the additional comparison / concatenation.

Reply:
So you mean `tagNameWithNamespace` is actually `originalTagName` or `rawTagName` which may or may not have name space?

Reply:
Yes, `tagNameWithNamespace` is the tag with a namespace (if it has a namespace).

I could use other names for that variable, such as `originalTagName` or `rawTagName` - whatever you'd prefer.

[As they say, naming things is one the hard problems of computer science](https://martinfowler.com/bliki/TwoHardThings.html)

Reply:
I completely agree and I faced it.

Reply:
I've updated this PR to use `rawTagName` as the name for this variable :)

Is there anything else I can do to help get this PR merged? I'm quite eager to use this fix in my project